src/metabase/util/embed.clj

```diff
@@ -108,6 +109,16 @@
   (or (get-in unsigned-token keyseq)
       (throw (ex-info (tru "Token is missing value for keypath {0}" keyseq) {:status-code 400}))))
 
+(defn maybe-populate-initially-published-at
+  "Populate `initially_published_at` if embedding is set to true"
+  [{:keys [enable_embedding initially_published_at] :as card}]
+  (cond
+    (and (true? enable_embedding) (nil? initially_published_at))
+    (assoc card :initially_published_at (now))
+
+    :else
+    card))
```

Comment:
Nice use of destructuring!

1. It's idiomatic to use `cond->` when we're optionally modifying something. That's what it's for
2. `card` should be `card-or-dashboard` to reflect its use with both
```suggestion
  [{:keys [enable_embedding initially_published_at] :as card-or-dashboard}]
  (cond-> card-or-dashboard
    (and (true? enable_embedding) (nil? initially_published_at))
    (assoc :initially_published_at (now))))
```

Reply:
Good calls! I didn't know `cond->` existed when I was writing it (of course lol) but that's pretty nice. I'll keep it for the future

Reply:
Oh, that `now` function returns a weird form: 
```
(now)
[:metabase.util.honey-sql-2/typed :%now {:database-type "timestamptz"}]
```

Instead of `(now)` you should use `(t/offset-date-time)`, which returns the current system time with an offset.
`t` is the alias for `java-time.api`, which you can require with `[java-time.api :as t]`.